aws-api/src/test/java/com/amplifyframework/api/aws/MultiAuthAppSyncGraphQLOperationTest.kt

```diff
@@ -0,0 +1,169 @@
+
+class MultiAuthAppSyncGraphQLOperationTest {
+
+    private val client = mockk<OkHttpClient>(relaxed = true)
+    private val onResponse = mockk<Consumer<GraphQLResponse<ModelWithTwoAuthModes>>>(relaxed = true)
+    private val onFailure = mockk<Consumer<ApiException>>(relaxed = true)
+    private val executorService = mockk<ExecutorService>(relaxed = true)
+    private val authTypes = mockk<AuthorizationTypeIterator>(relaxed = true)
+    private val request = ModelQuery[ModelWithTwoAuthModes::class.java, "1"]
+
+    private val responseFactoryMock = mockk<GsonGraphQLResponseFactory>()
+    private lateinit var operation: MultiAuthAppSyncGraphQLOperation<ModelWithTwoAuthModes>
+
+    @Before
+    fun setup() {
+        operation = MultiAuthAppSyncGraphQLOperation.Builder<ModelWithTwoAuthModes>()
+            .client(client)
+            .request(request)
+            .responseFactory(responseFactoryMock)
+            .onResponse(onResponse)
+            .onFailure(onFailure)
+            .apiRequestDecoratorFactory(mockk<ApiRequestDecoratorFactory>())
+            .executorService(executorService)
+            .endpoint("https://amazon.com")
+            .apiName("TestAPI")
+            .build()
+    }
+
+    // should submit dispatchRequest on auth-related error when more auth types are available
+    @Test
+    fun `submit dispatchRequest when more auth types available`() {
+
+        val body = "{\"errors\":" +
+            " [{\"message\": \"Auth error\"," +
+            " \"extensions\": {\"errorType\": \"Unauthorized\"}}]}"
+        val responseBody = body.toResponseBody()
+
+        val response = Response.Builder()
+            .code(200)
+            .body(responseBody)
+            .request(Request(url = "https://amazon.com".toHttpUrl()))
+            .protocol(Protocol.HTTP_1_0)
+            .message("testing for submit dispatch request when more auth types available")
+            .build()
+
+        val exception = ApiAuthException(
+            "Unable to successfully complete request with any of the compatible auth types.",
+            "Check your application logs for detail."
+        )
+
+        // Build the expected response.
+        val extensions: MutableMap<String, Any?> = HashMap()
+        extensions["errorType"] = "Unauthorized"
+        val gqlErrors = GraphQLResponse.Error("Unauthorized", null, null, extensions)
+        val gqlResponse = GraphQLResponse<ModelWithTwoAuthModes>(null, mutableListOf(gqlErrors))
+        gqlResponse.errors.replaceAll { gqlErrors }
+        every { responseFactoryMock.buildResponse<ModelWithTwoAuthModes>(any(), any(), any()) } returns gqlResponse
+
+        every { authTypes.hasNext() } returnsMany listOf(true, false)
+        operation.setAuthTypes(authTypes)
+
+        val runnableSlot = slot<Runnable>()
+        every { executorService.submit(capture(runnableSlot)) } answers {
+            runnableSlot.captured.run()
+            mockk(relaxed = true)
+        }
```

Comment:
Don't need slots for executing callbacks, can make it a little cleaner:

```suggestion
         every { executorService.submit(any()) } answers {
            firstArg<Runnable>().run()
            mockk()
        }
```